packages/pretty-format/src/__tests__/pretty-format-test.js

```diff
@@ -651,6 +651,35 @@ describe('prettyFormat()', () => {
       );
     });
 
+    it('supports a single element with custom React elements with props (using displayName)', () => {
+      function Cat() {
+        return React.createElement('div');
+      }
+      Cat.displayName = 'CatDisplayName';
+      assertPrintedJSX(
+        React.createElement('Mouse', {
+          prop: React.createElement(Cat, {foo: 'bar'}),
+        }),
+        '<Mouse\n  prop={\n    <CatDisplayName\n      foo="bar"\n    />\n  }\n/>',
+      );
+    });
+
+    it('supports a single element with custom React elements with props (using anonymous function)', () => {
+      assertPrintedJSX(
+        React.createElement('Mouse', {
+          /* eslint-disable prefer-arrow-callback */
+          prop: React.createElement(
+            function() {
```

Comment:
Since this is intentional (from our side) we can mock console:
```js
test('name', () => {
  const consoleError = console.error;
  console.error = jest.fn();
  // test code here...
  console.error = consoleError;
})
```